wakuv2/waku.go

```diff
@@ -535,38 +497,29 @@ func (w *Waku) runPeerExchangeLoop() {
 		case <-ticker.C:
 			w.logger.Info("Running peer exchange loop")
 
-			availablePeers := w.node.Host().Peerstore().Peers()
-			peersWithRelay := 0
-			for _, p := range availablePeers {
-				supportedProtocols, err := w.node.Host().Peerstore().SupportsProtocols(p, relay.WakuRelayID_v200)
-				if err != nil {
-					continue
-				}
-				if len(supportedProtocols) != 0 {
-					peersWithRelay++
-				}
-			}
-
-			peersToDiscover := w.cfg.DiscoveryLimit - peersWithRelay
-			if peersToDiscover <= 0 {
-				continue
-			}
-
 			// We select only the nodes discovered via DNS Discovery that support peer exchange
+			// We assume that those peers are running peer exchange according to infra config,
+			// If not, the peer selection process in go-waku will filter them out anyway
 			w.dnsAddressCacheLock.RLock()
+			var peers []peer.ID
 			for _, record := range w.dnsAddressCache {
 				for _, discoveredNode := range record {
 					if len(discoveredNode.PeerInfo.Addrs) == 0 {
 						continue
 					}
-					go w.identifyAndConnect(w.ctx, true, discoveredNode.PeerInfo)
-
+					// Attempt to connect to the peers.
+					// Peers will be added to the libp2p peer store thanks to identify
+					go w.connect(discoveredNode.PeerInfo, wps.DNSDiscovery)
```

Comment:
It shouldn't matter, because invoking connect on already connected peer doesn't do anything. 
It only does a dial if there is no connection to that peer. Reference libp2p code. 

https://pkg.go.dev/github.com/libp2p/go-libp2p@v0.32.2/core/host#Host.Connect